packages/react/src/products/hooks/useRecentlyViewedProducts.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  areRecentlyViewedProductsFetched,
+  areRecentlyViewedProductsLoading,
+  fetchRecentlyViewedProducts,
+  getRecentlyViewedProducts,
+  getRecentlyViewedProductsError,
+  getRecentlyViewedProductsPagination,
+  removeRecentlyViewedProduct,
+  saveRecentlyViewedProduct,
+} from '@farfetch/blackout-redux';
+import { useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction';
+import useProductListing from './useProductListing';
+import type { UseRecentlyViewedProductsOptions } from './types';
+
+const useRecentlyViewedProducts = (
+  productId: number,
+  options: UseRecentlyViewedProductsOptions = {},
+) => {
+  const { fetchQuery, enableAutoFetch = true, fetchConfig, save } = options;
+  const isLoading = useSelector(areRecentlyViewedProductsLoading);
+  const isFetched = useSelector(areRecentlyViewedProductsFetched);
+  const error = useSelector(getRecentlyViewedProductsError);
+  const products = useSelector(getRecentlyViewedProducts);
+  const pagination = useSelector(getRecentlyViewedProductsPagination);
+  const fetch = useAction(fetchRecentlyViewedProducts);
+  const reset = useAction(removeRecentlyViewedProduct);
+  const saveProduct = useAction(saveRecentlyViewedProduct);
+
+  const productIds = useMemo(
+    () =>
+      products
+        ?.reduce((acc, product) => {
+          if (product.productId !== productId) {
+            acc.push(product.productId);
+          }
+          return acc;
+        }, [] as number[])
+        .join('|'),
+    [productId, products],
+  );
+
+  const hasProducts = !!productIds;
+
+  const {
+    isLoading: isProductListLoading,
+    isFetched: isProductListFetched,
+    error: producListError,
+    data: { items } = {},
+  } = useProductListing('', {
+    enableAutoFetch: hasProducts,
+    setProductsListHash: false,
+    query: { id: productIds || ' ' },
+  });
+
+  useEffect(() => {
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
+      fetch(fetchQuery, fetchConfig);
+    }
+  }, [
+    enableAutoFetch,
+    error,
+    fetch,
+    fetchConfig,
+    fetchQuery,
+    isFetched,
+    isLoading,
+  ]);
+
+  useEffect(() => {
+    if (productId && save) {
+      saveProduct(productId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [productId, save]);
+
+  return {
+    isLoading: isLoading || !!isProductListLoading,
+    error: error || producListError,
+    isFetched: isFetched || isProductListFetched,
+    data: {
+      ...pagination,
+      products: items,
+    },
+    actions: {
+      fetch,
+      reset,
```

Comment:
I would not call this action reset, instead I would call I "remove" or "delete" to mirror the name given for the `save` action that I suggested in a previous comment.